Example/Example/AppFlowController.swift

```diff
@@ -0,0 +1,30 @@
+import UIKit
+import FlowControllerKit
+
+class AppFlowController {
+
+    let storyboard = UIStoryboard(name: "Main", bundle: nil)
+
+    var valueVC: ValueViewController!
+    var navigationController: UINavigationController
+
+    init(window: UIWindow) {
+        valueVC = storyboard.instantiate(ValueViewController.self)
+        navigationController = UINavigationController(rootViewController: valueVC)
+
+        window.rootViewController = navigationController
+        window.makeKeyAndVisible()
+    }
+
+    func start() {
+        let changeValueVC = storyboard.instantiate(ChangeValueViewController.self)
+        let modalVC = storyboard.instantiate(ModalViewController.self)
+
+        valueVC.changingValue = valueVC.showing(changeValueVC, preparation: changeValueVC.setup(value:))
```

Comment:
I think you've missed to type a value here? Or is it some syntactic sugar I don't know about? Either way it would be clear if something were typed there

Reply:
Okay, there is nothing missing ´valueVC.changingValue´ is a closure `valueVC.showing` is a method returning a closure but also takes a closure in the preparation `argument`. `changeValueVC.setup(value:)` is a reference to a method and we are not calling it here.

As all functions and methods in Swift are closure this works and all type information is inferred.